ov/storage_volume_template.go

```diff
@@ -0,0 +1,240 @@
+	ScopesURI                   utils.Nstring       `json:"scopesUri,omitempty"`
+	Name                        string              `json:"name,omitempty"`
+	State                       string              `json:"state,omitempty"`
+	Status                      string              `json:"status,omitempty"`
+	Type                        string              `json:"type,omitempty"`
+	URI                         utils.Nstring       `json:"uri,omitempty"`
+	InitialScopeUris            []utils.Nstring     `json:"initialScopeUris,omitempty"`
+	RootTemplateUri             utils.Nstring       `json:"rootTemplateUri,omitempty"`
+	StoragePoolUri              utils.Nstring       `json:"storagePoolUri,omitempty"`
+	Version                     string              `json:"version,omitempty"`
+	Uuid                        string              `json:"uuid,omitempty"`
+	Family                      string              `json:"family,omitempty"`
+	TemplateProperties          *TemplateProperties `json:"properties,omitempty"`
+}
+
+type TemplateProperties struct {
+	Name                          *TemplatePropertyDatatypeStruct     `json:"name,omitempty"`
+	StoragePool                   *TemplatePropertyDatatypeStruct     `json:"storagePool,omitempty"`
+	Size                          *TemplatePropertyDatatypeStructInt  `json:"size,omitempty"`
+	ProvisioningType              *TemplatePropertyDatatypeStruct     `json:"provisioningType,omitempty"`
+	DataTransferLimit             *TemplatePropertyDatatypeStruct     `json:"dataTransferLimit,omitempty"`
+	IsDeduplicated                *TemplatePropertyDatatypeStruct     `json:"isDeduplicated,omitempty"`
+	IsEncrypted                   *TemplatePropertyDatatypeStruct     `json:"isEncrypted,omitempty"`
+	IsPinned                      *TemplatePropertyDatatypeStruct     `json:"isPinned,omitempty"`
+	IopsLimit                     *TemplatePropertyDatatypeStruct     `json:"iopsLimit,omitempty"`
+	Folder                        *TemplatePropertyDatatypeStruct     `json:"folder,omitempty"`
+	TemplateVersion               *TemplatePropertyDatatypeStruct     `json:"templateVersion,omitempty"`
+	PerformancePolicy             *TemplatePropertyDatatypeStruct     `json:"performancePolicy,omitempty"`
+	VolumetSet                    *TemplatePropertyDatatypeStruct     `json:"volumeSet,omitempty"`
+	Description                   *TemplatePropertyDatatypeStruct     `json:"description,omitempty"`
+	IsAdaptiveOptimizationEnabled *TemplatePropertyDatatypeStructBool `json:"isAdaptiveOptimizationEnabled,omitempty"`
+	IsCompressed                  *TemplatePropertyDatatypeStruct     `json:"isCompressed,omitempty"`
+	DataProtectionLevel           *TemplatePropertyDatatypeStruct     `json:"dataProtectionLevel,omitempty"`
+	IsShareable                   *TemplatePropertyDatatypeStructBool `json:"isShareable,omitempty"`
+}
+
+type TemplatePropertyDatatypeStruct struct {
+	Meta        Meta          `json:"meta,omitempty"`
```

Comment:
Can these Meta be pointer in all the structs used

Reply:
Changed it.